packages/nodejs/src/cli/diagnose.ts

```diff
@@ -341,6 +337,80 @@ export class Diagnose {
     }
   }
 
+  printConfiguration(
+    {options, sources}: {
+      options: { [key: string]: any },
+      sources: { [source: string]: { [key: string]: any } }
+    }
+  ) {
+    console.log(`Configuration`)
+    Object.keys(options)
+      .sort()
+      .forEach(key => {
+        let keySources = this.configurationKeySources(key, sources)
+
+        switch (Object.keys(keySources).length) {
+          case 0:
+            if (key == 'log_file_path') {
+              // The `log_file_path` configuration key is generated after
+              // the config is read and does not belong to any sources.
+              // Since it is derived from `log_path`, let's give it the
+              // same source as `log_path`'s final value.
```

Comment:
We should not set this key on the config object. See also my comment here: https://github.com/appsignal/integration-guide/issues/78#issuecomment-960716072

That way we don't have to filter it out again here. I don't want to print a different value here, that sounds confusing.

Reply:
> We should not set this key on the config object.

Agreed 👍 I'll look into making this change. It's the only value that works in this way, so by removing `log_file_path` from the config, we could remove `case 0:` here entirely, or throw an error if it were to happen.

> I don't want to print a different value here, that sounds confusing.

It's not printing a different value, just pretending the value comes from the same source as wherever `log_path` comes from. So, if `log_path` was overriden on the environment variables, the output would be:

```
  log_path: /custom/location
    Sources:
      default: /tmp
      env: /custom/location
  log_file_path: /custom/location/appsignal.log (Loaded from: env)
```

Without this, it would not say `Loaded from: env`, implying it was set from `default`, which I thought would be more confusing.